test/common/config/subscription_factory_test.cc

```diff
@@ -190,18 +188,26 @@ INSTANTIATE_TEST_CASE_P(SubscriptionFactoryTestApiConfigSource,
                                           envoy::api::v2::core::ApiConfigSource::REST,
                                           envoy::api::v2::core::ApiConfigSource::GRPC));
 
+// TODO(jbuckland) add test case for ApiConfigSource::GRPC
+
 TEST_P(SubscriptionFactoryTestApiConfigSource, NonExistentCluster) {
   envoy::api::v2::core::ConfigSource config;
   auto* api_config_source = config.mutable_api_config_source();
   api_config_source->set_api_type(GetParam());
   api_config_source->add_cluster_names("eds_cluster");
   Upstream::ClusterManager::ClusterInfoMap cluster_map;
   EXPECT_CALL(cm_, clusters()).WillOnce(Return(cluster_map));
-  EXPECT_THROW_WITH_MESSAGE(subscriptionFromConfigSource(config)->start({"foo"}, callbacks_),
-                            EnvoyException,
-                            "envoy::api::v2::core::ConfigSource must have a statically defined "
-                            "non-EDS cluster: 'eds_cluster' "
-                            "does not exist, was added via api, or is an EDS cluster");
+  if (api_config_source->api_type() == envoy::api::v2::core::ApiConfigSource::GRPC) {
+    EXPECT_THROW_WITH_MESSAGE(
```

Comment:
I think for `subscription_factory_test`, we should go in and add additional tests to cover all the branches possible now in checkApiConfigSourceSubscriptionBackingCluster

Reply:
Done in d149e7f.